src/xrpld/app/tx/detail/PermissionedDomainSet.cpp

```diff
@@ -0,0 +1,158 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2024 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <xrpld/app/tx/detail/PermissionedDomainSet.h>
+#include <xrpld/ledger/View.h>
+#include <xrpl/protocol/STObject.h>
+#include <xrpl/protocol/TxFlags.h>
+#include <map>
+#include <optional>
+
+namespace ripple {
+
+NotTEC
+PermissionedDomainSet::preflight(PreflightContext const& ctx)
+{
+    if (!ctx.rules.enabled(featurePermissionedDomains))
+        return temDISABLED;
+    if (auto const ret = preflight1(ctx); !isTesSuccess(ret))
+        return ret;
+
+    auto const credentials = ctx.tx.getFieldArray(sfAcceptedCredentials);
+    if (credentials.empty() || credentials.size() > PD_ARRAY_MAX)
+        return temMALFORMED;
+
+    return preflight2(ctx);
+}
+
+XRPAmount
+PermissionedDomainSet::calculateBaseFee(ReadView const& view, STTx const& tx)
+{
+    if (tx.isFieldPresent(sfDomainID))
```

Comment:
@dangell7 including the domain means updating an existing record. Without the domain means creating a new record. Regarding whether or not an extra fee should be charged (in addition to owner reserve), @mvadari, what do you think? Should there be an extra fee for creating a PermissionedDomain object, in addition to owner reserve?

From what I can tell, the following transactions require an extra fee like this:
AMMCreate, LedgerStateFix, DeleteAccount

Otherwise, things that create objects and increment the owner count, don't seem to also charge an extra fee. Such as OfferCreate.

How should it be for this object? Either way, it's a simple fix but should probably be part of the spec, please.

Reply:
The examples you bring up all are special cases (for example, `AMMCreate` is creating an account, not just a normal object). Creating a normal object (like `PermissionedDomain`) just requires locking up reserve, not an extra fee. It's not specified in the spec because it's not a special situation - it's a standard transaction with a standard transaction fee.

Reply:
fixed